src/irvisualizer/README.md

```diff
@@ -0,0 +1,32 @@
+# Javascript toolchain for IRVisualizer: the single page web application
+
+**Features:**
+
+* To trigger Github's online Dependabot alerts for security vulnerabilities.
+
+* To enable static analyzer for the Javascript/ES6 language.
+
+* To visualize the Halide IR and the assembly code interactively
+with a web browser.
+
+**Usage:**
+
+* Execute `make depend` to scan for broken Javascript dependencies.
+
+* Execute `make lint` to scan the main script for potential syntax errors and
+  bugs.
+
+**TODO**
+
+* [ ] Rename the extensions from `*.template.html` to `template/*.[js|css|html]`.
+
+* [ ] Modernize the main script to ES6.
+
+* [ ] Import all dependencies with the `import` statement, not via the `<script>` tag.
+
+* [ ] Resolve all static analyzer warnings, e.g. "buttonHide" is already defined, "btns.size()" is undefined, etc.
+
```

Comment:
From @steven-johnson :

ok, after doing brew install node yarn eslint esbuild on my Mac laptop, doing make lint results in:

```
$ make lint
yarn run lint
yarn run v1.22.19
$ eslint StmtToViz_javascript.template.html

/Users/srj/GitHub/Halide/src/irvisualizer/StmtToViz_javascript.template.html
  1:1  error  Parsing error: Unexpected token <

✖ 1 problem (1 error, 0 warnings)

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
make: *** [lint] Error 1
```

Yep, it is an TODO item: refactor the IRVisualizer code base to allow auto-formatting with `clang-format`, and then static analyzer checking with `eslint`. What you see with `Unexpected token <` is expected.

Reply:
It's not at all clear that `make lint` is currently **expected** to fail -- I'd assume that all the build targets in the Makefile do what you expect. Please add appropriate comments and/or documentation -- I realize this may seem obvious to you but it certainly wasn't to me, so I suspect it may be non-obvious to others as well.

Reply:
Done moving the draft `Makefile` contents to Readme.md . Thank you for the guidance.